portindex2json/portindex2json.tcl

```diff
@@ -91,4 +91,13 @@ while {[gets $fd line] >= 0} {
 }
 
 chan configure stdout -encoding utf-8
-puts [::json::write array {*}$objects]
+if {$argc == 2} {
+    # If a commit hash is provided then the output is a JSON object with keys "commit" and "ports".
+    # Value of "ports" is a list of JSON objects, each object containing info for one port.
+    set json_output(commit) [::json::write string [lindex $argv 1]]
+    set json_output(ports) [::json::write array {*}$objects]
+    puts [::json::write object {*}[array get json_output]]
+} else {
```

Comment:
Doing it as an `if-else` and end up with a completely different file when additional info is missing sounds really strange. You should output nearly the same information, only with that additional information missing.

Reply:
Thank you so much. The output definitely needed to be consistent in all cases. If the `-info` argument is not supplied now, the key `"info"` will be dropped and only `"ports" will be in the output.